mobile/src/people/people-department.tsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import { Action } from 'redux';
+import { connect, Dispatch } from 'react-redux';
+import { FlatList, Text, View, StyleSheet, ListRenderItemInfo } from 'react-native';
+
+import { EmployeesList } from './employees-list';
+import { AppState } from '../reducers/app.reducer';
+import { PeopleActions, navigatePeopleDepartment } from '../reducers/people/people.action';
+import { loadEmployeesForDepartment } from '../reducers/organization/organization.action';
+import { EmployeesStore, EmployeeMap, EmployeeIdsGroupMap } from '../reducers/organization/employees.reducer';
+
+interface PeopleDepartmentProps {
+    employeesMap: EmployeeMap;
+    departmentId: string;
+    employeesSubsetFilterCallback: any;
+}
+
+interface PeopleDepartmentDispatchProps {
+    navigatePeopleDepartment: (departmentId: string) => void;
+}
+
+const mapStateToProps = (state: AppState): PeopleDepartmentProps => ({
+    employeesMap: state.organization.employees.employeesById,
+    departmentId: state.userInfo.employee.departmentId,
+    employeesSubsetFilterCallback: state.people.employeesDepartmentSubsetFilterCallback
+});
+
+const mapDispatchToProps = (dispatch: Dispatch<PeopleActions>) => ({
+    navigatePeopleDepartment: (departmentId: string) => { 
+        dispatch(navigatePeopleDepartment(departmentId)); 
+        dispatch(loadEmployeesForDepartment(departmentId));
```

Comment:
the second dispatch should be called in response to the first one, something like action.ofType('NAVIGATE_DEPARTMENT).map(x => loadEmployeesForDepartment(x.departmentId))